### ISSUE TITLE
Make a test unit for multi turret

### ISSUE THREAD

**Issue by methusalah** [methusalah/OpenRTS#20]
deux tourelles, si possible éloignées, avec une arme sur chaque. Par exemple un truc comme le mothership des protoss, ou un cube avec deux cube et des cubes. bref, un machin pour tester le multi turrets quoi

**Comment by roroah** [comment]
Je comprend pas, le mothership des protoss ressemble en rien en la description (aka: deux tourelles, si possible éloignées, avec une arme sur chaque. )

Faut un mothership ou deux tourelle ?

**Comment by methusalah** [comment]
Deux tourelle, juste pour tester

**Comment by roroah** [comment]
J'ai commencer la premiere tourelle, mais j'arrive pas a la mettre dans le moteur pour voir ce que sa donne en terme de scale et autre, la parti du wiki concernant l'import de tourelle est manquante ou sinon https://github.com/methusalah/OpenRTS/wiki/Guide-for-dataset-contribution#turret est incomplet, ou sinon je sais pas lire et je m'escuse :p

Si possible me pointer dans la bonne direction pour inplanter la tourelle.

Note à part vis à vis du wiki :
Faudrais expliquer les differentes valeur possible et a quoi correspond quoi, je peux ouvrir un issue concernant le wiki sinon, un milestone serais peut-etre plus adapter quand même.
exemple à https://github.com/methusalah/OpenRTS/wiki/Guide-for-dataset-contribution#unit : 
<MoverLink value="GroundProne"/>
- Sa correspond a quoi "MoverLink" ?
- "GroundProne" c'est une valeur arbitraire ou une valeur parmis un set ?

**Comment by methusalah** [comment]
Indeed, the turret chapter of the wiki is incomplete. The turret rotation is made around the named bone of the unit model's skeleton.

When you create a unit, you attach a ModelActor. The model must have a bone to rotate, corresponding to the turret rotation axis. It's that bone that the turret names with the attribute "BoneName".

Then, you can specify weapons with or without attached turret. If a weapon has a turret, it will ask this turret to rotate to reach the target. Else it will ask the unit.

note that the turret rotation is only around Z axis.

I created a milestone for the wiki issues

**Comment by roroah** [comment]
You lost me, i though turret was something like static turret !? (like : [1])
If not, I'll suppose that turret are what the current tank use as weapon which in case is a weapon extending from turret.
About the weapon having attached turret or not, it is pretty confusing, a turret have weapon, not the opposite.


[1] http://3drt.com/store/images/detailed/2/wargear-sci-fi-lowpoly-3d-turrets_19.jpg

**Comment by methusalah** [comment]
Sorry for the misunderstanding. The turret is the mobile part of units that
a weapon can rotate instead of the whole unit itself.

Mechanically you're right, a turret holds the weapon. But in the game
logic, we construct an abstract weapon that may control a turret, or not.

**Comment by roroah** [comment]
so if i recap, what is needed is what the tank already.
So an unit holding multiple turret, somethink like an upgraded tank.

**Comment by methusalah** [comment]
Yes!

**Comment by roroah** [comment]
Ok, I'll see what i can get for, the unit can be air ? or must be ground ?

**Comment by methusalah** [comment]
It doesn't matter for the test
